src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -327,6 +323,24 @@ public boolean isNameVisibleTo(
         }
         return false;
     }
+
+    /**
+     * Checks if the user gave the response, or is part of the team that gave the response.
+     */
+    private boolean isUserGiverOfResponse(FeedbackQuestionAttributes question,
+            FeedbackResponseAttributes response, String userEmail, CourseRoster roster) {
+        if (question.giverType == FeedbackParticipantType.TEAMS) {
+            // check if user is part of team if responses are given by teams
+            if (roster.isStudentsInSameTeam(response.giver, userEmail)) {
+                return true;
+            }
+        } else {
+            if (response.giver.equals(userEmail)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
This can be:
```java
return question.giverType == FeedbackParticipantType.TEAMS && roster.isStudentsInSameTeam(response.giver, userEmail)
        || question.giverType != FeedbackParticipantType.TEAMS && response.giver.equals(userEmail);
```